src/js/oncoprintmodel.js

```diff
@@ -1277,19 +1284,36 @@ var OncoprintModel = (function () {
 				cluster_input[id] = cluster_input[id] || {};
 				cluster_input[id][track_id] = value;
 			}
-		}
-		//do hierarchical clustering in background:
-		var self = this;
+	}
+	if (!Object.keys(cluster_input).length) {
+	    // skip clustering if there's nothing to cluster
+	    return;
```

Comment:
return a resolved promise  - see below for what the format should be

Reply:
Well spotted — I'd completely missed that!